spherical_geometry/tests/test_basic.py

```diff
@@ -536,3 +540,35 @@ def test_math_util_angle_nearly_coplanar_vec():
 
     assert_allclose(angles[:-1], np.pi, rtol=0, atol=1e-16)
     assert_allclose(angles[-1], 0, rtol=0, atol=1e-32)
+
+
+def test_inner1d():
+    vectors = [
+        [1.0, 1.0, 1.0],
+        3 * [1.0 / np.sqrt(3)],
+        [1, 0.5, 1],
+        [1, 0.15, 1],
+        [-1, 0.1, -1]
+    ]
+    lengths = great_circle_arc._inner1d_np(vectors, vectors)
+
+    assert_allclose(lengths, [3.0, 1.0, 2.25, 2.0225, 2.01], rtol=0, atol=1e-15)
+
+
+@pytest.mark.skipif(math_util is None, reason="math_util C-ext is missing")
+def test_math_util_inner1d():
+    vectors = [
+        [1.0, 1.0, 1.0],
+        3 * [1.0 / np.sqrt(3)],
+        [1, 0.5, 1],
+        [1, 0.15, 1],
+        [-1, 0.1, -1]
+    ]
+    lengths = great_circle_arc._inner1d_np(vectors, vectors)
+
+    assert_allclose(
+        lengths,
+        math_util.inner1d(vectors, vectors),
+        rtol=0,
+        atol=1e-15
+    )
```

Comment:
These can be combined to avoid code duplication:

```suggestion
def test_inner1d():
    vectors = [
        [1.0, 1.0, 1.0],
        3 * [1.0 / np.sqrt(3)],
        [1, 0.5, 1],
        [1, 0.15, 1],
        [-1, 0.1, -1]
    ]
    lengths = great_circle_arc._inner1d_np(vectors, vectors)

    assert_allclose(lengths, [3.0, 1.0, 2.25, 2.0225, 2.01], rtol=0, atol=1e-15)

    if math_util is None:  # math_util C-ext is missing
        return

    assert_allclose(
        lengths,
        math_util.inner1d(vectors, vectors),
        rtol=0,
        atol=1e-15
    )
```

Reply:
it could but then it will not be reported that one test was skipped.

Reply:
I have modified `test_math_util_inner1d` a little (to just check the values; just like the `test_inner1d` test). Now `test_math_util_inner1d` and `test_inner1d` are similar tests. I will keep both tests as they test two different functions.